### PR TITLE
275 lunch ticket qr code

### LINKED ISSUE #275: Lunch ticket qr code
An ARKAD-funktionär (volunteer) should be able to see the QR code of their lunch ticket when logged in to the app. A button could be added to the profile page which displays the code when clicked.

### PR DESCRIPTION
**Pull request by emil-ronkainen**
For any admin, volunteer or student, there is now a button on the profile page for displaying their lunch ticket as a qr code.

Closes #275.

## Checklist
- [x] Have you remembered to run the `test_actions.sh` script?
- [x] Have you rebased the branch on the latest main?
- [x] Have you tested the change locally?

### CODE REVIEW

**Comment by Stronkness** [comment]
Note that students shouldn't see lunch or banquett tickets in the app as they are not a part of ARKAD. This need to be fixed.
For now only admins and volounteers should see the ticket, but company representative can be included also as this will be IT 24s job

File: screens/profile/ProfileScreen.tsx [hunk 1]
```diff
@@ -99,6 +112,17 @@ export default function ProfileScreen({ navigation }: ProfileScreenParams) {
             editingProfile={false}
             onPress={() => navigation.navigate("ProfileSwitchScreen", { screen: "edit", id: 0 })}
           />
+          { !company && (
```
Comment (Stronkness) [inline screens/profile/ProfileScreen.tsx]: !student right?

**Review by Stronkness — CHANGES_REQUESTED** [review]
Change to !student

**Review by emil-ronkainen — COMMENTED** [review]
Updated conditional rendering of ShowLunchTicketButton. Now displays for every user but the student role

**Comment by emil-ronkainen** [comment]
After some testing it seems that this issue might already be resolved. Due to the lunch tickets using the same "pipeline" as any other event, it is already displayed on the profile page and can be clicked to open a view with further information and the qr code. Attatching an image of the situation below (running on branch "mockDataForTicketAndEVEnt" on backend and logged in as user "student1@example.com")

I am marking the PR as draft until this is addressed.

image

**Comment by Stronkness** [comment]
Honestly i think this looks great, i think we should stick with the pipeline event. Fix the merge conflict and we good to go

**Comment by emil-ronkainen** [comment]
Closing this as the feature is already implemented

**Pull request closed without merging**